### PR TITLE
Remove validation done in libcalico

### PR DESCRIPTION
**Pull request by L-MA**
Wait for https://github.com/projectcalico/libcalico/pull/53 to get it

### CODE REVIEW

File: calico_kubernetes/policy.py [hunk 1]
```diff
@@ -98,34 +98,24 @@ def _generate_rule(self, arguments):
         if protocol:
             rule_args["protocol"] = protocol
         if src_ports:
-            rule_args["src_ports"] = self._validate_ports(src_ports)
+            rule_args["src_ports"] = src_ports
```
Comment (caseydavenport) [inline calico_kubernetes/policy.py]: I think we still need something here to convert to a list, correct?

`src_ports` is a string which looks like this: `"10,20,30"` but it needs to looks like this: `[10, 20, 30]`

File: calico_kubernetes/policy.py [hunk 2]
```diff
@@ -98,33 +98,36 @@ def _generate_rule(self, arguments):
...
+    def _sanitize_ports(self, value):
         """
-        Validates the given value can be parsed as an int.
-        TODO: Move validation code into libcalico
+        Splits the given string of ports, delimited by a comma, into a list.
+        Removes all whitespace before splitting.
+        Validation is done in libcalico.
+
+        :param value: Port string.  E.g. "80,8001"
+        :return: List of ports.  E.g [80, 8001]
```
Comment (caseydavenport) [inline calico_kubernetes/policy.py]: List of ports is actually a list of strings, so represent it like ["80", "8001"]

File: calico_kubernetes/policy.py [hunk 3]
```diff
@@ -98,33 +98,36 @@ def _generate_rule(self, arguments):
...
         """
-        try:
-            return int(value)
-        except ValueError:
-            raise PolicyParseError("Failed to parse %s, expecting int", value)
+        value = value.replace(' ', '')
+        ports = value.split(",")
```
Comment (caseydavenport) [inline calico_kubernetes/policy.py]: Something like this would be more Pythonic:
```
ports = [s.strip() for s in value.split(",")]
```

Comment (caseydavenport) [inline calico_kubernetes/policy.py]: You could probably get rid of this whole function and inline this up above -

```
rule_args["src_ports"] = [s.strip() for s in value.split(",")]
```

File: calico_kubernetes/policy.py [hunk 4]
```diff
@@ -98,33 +98,36 @@ def _generate_rule(self, arguments):
...
+        Removes all whitespace before splitting.
+        Validation is done in libcalico.
+
+        :param value: Port string.  E.g. "80,8001"
+        :return: List of ports.  E.g [80, 8001]
         """
-        try:
-            return int(value)
-        except ValueError:
-            raise PolicyParseError("Failed to parse %s, expecting int", value)
```
Comment (caseydavenport) [inline calico_kubernetes/policy.py]: I'm wondering about the PolicyParseError now.  `libcalico` throws ValueErrors, so I wonder if we should be doing the same to be consistent.

At a minimum, PolicyParseError should inherit from ValueError.

File: calico_kubernetes/tests/policy_test.py [hunk 5]
```diff
@@ -135,18 +135,15 @@ def test_parse_icmp_rule(self):
         # Assert on the result
         assert_equal(parsed['action'], "allow")
         assert_equal(parsed['protocol'], "icmp")
-        assert_equal(parsed['icmp_type'], 8)
+        assert_equal(parsed['icmp_type'], '8')
         assert_equal(parsed['src_net'], IPNetwork("1.2.0.0/16"))
 
     @parameterized.expand([
         ("allow icmp type code 4"),
-        ("allow icmp type ten code 4"),
```
Comment (caseydavenport) [inline calico_kubernetes/tests/policy_test.py]: why remove these test cases?

Comment (L-MA) [inline calico_kubernetes/tests/policy_test.py]: calico_kubernetes uses a specific version of libcalico, not master, therefore, the libcalico validation changes are not reflected in these tests. Should we remove the version locking of libcalico or should we remove these test cases?

Comment (caseydavenport) [inline calico_kubernetes/tests/policy_test.py]: Yep, we should probably be stuck on master libcalico.

**Comment by caseydavenport** [comment]
@L-MA - when you've got a chance, can you bring this guy up to date?

File: calico_kubernetes/policy.py [hunk 6]
```diff
@@ -98,73 +98,35 @@ def _generate_rule(self, arguments):
...
-
     def _validate_label(self, value):
         """
         Takes the given label, validates it, and returns the equivalent tag.
         """
         match = LABEL_REG.search(value)
         if not match:
-            raise PolicyParseError("Failed to parse %s, expecting "
+            raise ValueError("Failed to parse %s, expecting "
                                    "label of form X=Y", value)
```
Comment (caseydavenport) [inline calico_kubernetes/policy.py]: Awkward indentation here

**Comment by caseydavenport** [comment]
One more comment to address then LGTM

**Pull request merged**